.travis.yml

```diff
@@ -1,38 +1,51 @@
 ---
-language: node_js
-node_js:
-  - "6"
+jobs:
+  include:
+    - stage: lint
+      language: python
+      python: 3.6
+      before_install:
+        - export PATH=$PATH:./node_modules/.bin
+        - export NODE_PATH=$NODE_PATH:./node_modules
+        - nvm install
+        - npm install
+        - npm install eslint remark remark-lint remark-cli
+        - pip3 install coala coala-bears
+      script:
+        - coala --non-interactive -V
 
-sudo: false
+    - stage: npm_test_and_deploy
+      language: node_js
+      node_js:
+        - "6"
+      cache:
+        directories:
+          - $HOME/.npm
+          - $HOME/.cache  # includes bowers cache
 
-cache:
-  directories:
-    - $HOME/.npm
-    - $HOME/.cache # includes bowers cache
+      before_install:
+        - npm config set spin false
+        - npm install -g bower phantomjs-prebuilt
+        - bower --version
+        - phantomjs --version
 
-before_install:
-  - npm config set spin false
-  - npm install -g bower phantomjs-prebuilt
-  - bower --version
-  - phantomjs --version
+      install:
+        - npm install
```

Comment:
There is no global `install` section on Travis to run `npm install` on all stage?

Reply:
Learning Travis is time well spent; your project relies on it.